torchaudio/models/wavernn.py

```diff
@@ -324,3 +334,37 @@ def forward(self, waveform: Tensor, specgram: Tensor) -> Tensor:
 
         # bring back channel dimension
         return x.unsqueeze(1)
+
+
+def _wavernn(arch: str, pretrained: bool, progress: bool, **kwargs: Any) -> WaveRNN:
+    model = WaveRNN(**kwargs)
+    if pretrained:
+        state_dict = load_state_dict_from_url(model_urls['wavernn'],
+                                              progress=progress)
+        model.load_state_dict(state_dict)
+    return model
+
+
+def wavernn_10k_epochs_8bits_ljspeech(pretrained: bool = True, progress: bool = True, **kwargs: Any) -> WaveRNN:
```

Comment:
Note: `pretrained: bool = True` is not future proof.
In case there is a newer pretrained parameters, and we want to be able to have both parameters available, this signature makes it difficult to have data backward compatibility, friendly UX for picking the right data, and maintainability. It is better to define some ID which uniquely determines which pretrained data it should fetch.

Reply:
I feel that arbitral `kwargs` and loading pretrained model (`pretrained = True`) do not work well together.
There should be a check that prevents users from providing `kwargs` that is not compatibile with pretrained model in the case `pretrained=True`, or simply separate the interface for fetching pretrained model and building un-trained model.

Reply:
What does `10k_epochs` mean when `pretrained = False`?

Reply:
We are actually following the convention from [torchvision](https://github.com/pytorch/vision/blob/master/torchvision/models/vgg.py). In the future, if there are new model (or new pretrained weights), we can just add a new function. In their example, they have separate functions for newer pretrained parameters Vgg11, Vgg13, Vgg16, ... and so on.

I also like what you have suggested, which seems to be changing the `pretrained` parameter into taking a string that matches corresponding weight URL. But the downside is that it is different from torchvision, and people maybe more familiar with their convention.

> What does `10k_epochs` mean when `pretrained = False`?
good point, `10k_epochs` don't mean anything when `pretrained = False`.

@vincentqb Please let me know your comment on which interface to take, the one that follows torchvision or using the `pretrained` parameter to take in some ID for each weight URL. Also please let me know if I missed anything :)

Reply:
Notice that `torchvision`'s code has some legacy/historic constraints and IIRC, @datumbox had some problem with it. There was a model code change that originated from the upstream PyTorch core, but there was no way to adopt the change without making BC-breaking. If we copy them as-is, we will go through the same issue in future. We can take the inspiration from `torchvision`, but sticking to it is a different concern.

Reply:
I've changed the interface to (get_pretrained_wavernn)[https://github.com/pytorch/audio/pull/1612/files#diff-7f4ad259da182d3840fe4c75dd0c0b7b9fe30f31503df0eff87b1e5c9378cac4R352].

Please let me know what you think. :)

Reply:
@mthrok Apologies for not seeing this earlier. I see that the final PR adopts a different approach that the one proposed in this commit. Indeed even in TorchVision we are in the process of assessing better support of versioning. I'll get to it once I'm back.

Out of curiosity, I see that the adopted solution does not permit for creating a model of specific configuration WITHOUT pre-trained weights. Is this something you would like to support on the future? Aka load the weight with specific architecture config/params but without loading pre-trained weights.

Reply:
>  Is this something you would like to support on the future?

@datumbox I am open to it if there is a feature request with valid use cases.
It feels to me that, it is just about passing the fixed configuration to the constructor of `WaveRNN` class, adding that feature to this factory function is nothing but a syntax sugar. So it's nice if we have a good justification for the syntax sugar first. In the case of torchvision's ResNet, there are bunch of different yet popular configurations across different tasks, so I think it's well justified. For WaveRNN, I am not sure a.t.m. And I think it's not too late to wait for someone with expertise in TTS domain to show up and advise us what are the common use cases.

Reply:
Sounds good. Can't comment on what's best on the Audio domain as my knowledge is limited. I'll take your input into account for the proposed solution on vision to see if we can converge to a cross domain solution.

Reply:
Cool. just FYI in case of TTS or STT, we always want to have the possibility of multiple pre-trained models for the same architecture because in simple case STT models are trained per language.

BTW: If I were to add that feature now, I think I would make a signature like the following;

```pyton
def wave_rnn(checkpoint=None, config=None) -> Module:
    if checkpoint is not None and config is None:
        # return pre-trained
    if checkpoint is None and config is not None:
        # return un-trained
    raise ValueError("One of (and only one of) `checkpoint` or `config` must be given.")
 ```

Reply:
There are pros/cons with having different params for checkpoint and config. It's a lengthy topic I don't want to open here cause it's going to get buried. I think we should catch up once I am back to discuss options and get your thoughts.